docs/CODING_STYLE.md

```diff
@@ -152,25 +152,35 @@ layout: default
 ## Using C Constructs
 
 - Allocate local variables where it makes sense: at the top of the block, or at
-  the point where they can be initialized. `r` is typically used for a local
-  state variable, but should almost always be declared at the top of the
+  the point where they can be initialized. Avoid huge variable declaration
+  lists at the top of the function. In long functions that require many
+  variables, consider moving some of the declarations close to the point of
+  first use. As an exception, `r` is typically used for a local state variable,
+  but should almost always be declared as the last variable at the top of the
   function.
 
   ```c
   {
-          uint64_t a, b;
+          uint64_t a;
           int r;
 
-          a = frobnicate();
-          b = a + 5;
+          r = frobnicate(&a);
+          if (r < 0)
+                  …
+
+          uint64_t b = a + 11, c;
```

Comment:
i really don't like this style I must say. you mix a declaration with code, and then another declartion. I'm, not convinced the idea of stuffing so much in one line is really the best thing.

I am pretty sure we should instead say: at most one code expression per line, and optionally prefix that with one variable declaration if you must.

i.e. this is bad I think:

```c
int a = foo(), b = bar() * 4711 + 888, c = k + 49;
```

And this should be written as:

```c
int a = foo();
int b = bar() * 4711 + 888;
int c = k + 49;
```

Or lennart would prefer:

```
int a, b, c;
a = foor();
b = bar() * 4711 + 88;
c = k + 49;
```

Reply:
I think we're pretty much in agreement. I find the second or third options much nicer than the first. And I think the proposed text explicitly discourages that: "do not mix multiple variable definitions with non-trivial function invocations in the block at the top of the function". The qualifier of "in the block at the top of the function" was stupid — I didn't mean it as "do it in other places", but rather "don't do it especially there". But it's better to drop it.

Whether option two or three is nicer depends on the context… If all three variables are `int`, I think either is fine. If they have different types, than option two would be nicer I think:
```C
uint64_t a;
int32_t b;
char *c;
a = foo();
b = bar() * 5;
c = get_some_string();
```
I agree that it's best if we leave the text neutral.